build-system/tasks/visual-diff.js

```diff
@@ -35,10 +100,463 @@ function setPercyBranch() {
+  const {body} = await request(statusUri, {json: true}).catch(error => {
+    log('fatal', 'Failed to query Percy build status:', error);
+  });
+  return body;
+}
+
+/**
+ * Waits for Percy to finish processing a build.
+ * @param {string} buildId ID of the ongoing Percy build.
+ * @return {!JsonObject} The eventual status of the Percy build.
+ */
+async function waitForBuildCompletion(buildId) {
+  log('info', 'Waiting for Percy build', colors.cyan(buildId),
+      'to be processed...');
+  const startTime = Date.now();
+  let lastProgressUpdateTime = Date.now();
+  let status;
+  do {
+    status = await getBuildStatus(buildId);
+
+    if (Date.now() - lastProgressUpdateTime >=
+            BUILD_PROCESSING_PROGRESS_REPORT_MS) {
+      log('info', 'Still waiting for Percy build', colors.cyan(buildId),
+          'to be processed...');
+      lastProgressUpdateTime = Date.now();
+    }
+
+    await sleep(BUILD_PROCESSING_POLLING_INTERVAL_MS);
+  } while (status.state != 'finished' && status.state != 'failed' &&
+               Date.now() - startTime < BUILD_PROCESSING_TIMEOUT_MS);
+  return status;
+}
+
+/**
+ * Verifies that a Percy build succeeded and didn't contain any visual diffs.
+ * @param {!JsonObject} status The eventual status of the Percy build.
+ * @param {string} buildId ID of the Percy build.
+ */
+function verifyBuildStatus(status, buildId) {
+  switch (status.state) {
+    case 'finished':
+      if (status.total_comparisons_diff > 0) {
+        if (MASTER_BRANCHES_REGEXP.test(status.branch)) {
+          // If there are visual diffs on master or a release branch, fail
+          // Travis. For master, print instructions for how to approve new
+          // visual changes.
+          if (status.branch == 'master') {
+            log('error', 'Found visual diffs. If the changes are intentional,',
+                'you must approve the build at',
+                colors.cyan(`${PERCY_BUILD_URL}/${buildId}`),
+                'in order to update the baseline snapshots.');
+          } else {
+            log('error', `Found visual diffs on branch ${status.branch}`);
+          }
+        } else {
+          // For PR branches, just print a warning since the diff may be into
+          // intentional, with instructions for how to approve the new snapshots
+          // so they are used as the baseline for future visual diff builds.
+          log('warning', 'Percy build', colors.cyan(buildId),
+              'contains visual diffs.');
+          log('warning', 'If they are intentional, you must first approve the',
+              'build at', colors.cyan(`${PERCY_BUILD_URL}/${buildId}`),
+              'to allow your PR to be merged.');
+          log('warning',
+              'You must then wait for your merged PR to be tested on master,',
+              'and approve the next "master" build at',
+              colors.cyan(PERCY_BUILD_URL),
+              'in order to update the visual diff baseline snapshots.');
```

Comment:
This specific line is no longer accurate, since we switched to automatic baseline selection. Once a build successfully generates all snapshots on master, it will be used as the new baseline whether or not it contains any diffs relative to the previous baseline.

I think you can simply delete lines 216-220.